csharp/test/Drivers/Snowflake/ValueTests.cs

```diff
@@ -0,0 +1,257 @@
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.Parse("-10")));
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.Parse("10")));
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.MinValue));
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.MaxValue));
+        }
+
+        /// <summary>
+        /// Validates if driver can handle a large scale Number type correctly
+        /// </summary>
+        [SkippableFact]
+        public void TestSmallScaleNumberRange()
+        {
+            string columnName = "SMALLSCALENUMBER";
+            string table = CreateTemporaryTable(string.Format("{0} NUMBER(38,2)", columnName));
+            ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, 0);
+            ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.Parse("4.85"));
+            ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.Parse("-999999999999999999999999999999999999.99"));
+            ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.Parse("999999999999999999999999999999999999.99"));
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.MinValue));
+            Assert.Throws<AdbcException>(() => ValidateInsertSelectDeleteSingleDecimalValue(table, columnName, SqlDecimal.MaxValue));
+
+            // Snowflake allows insert when input goes beyond scale, but storage will round value.
+            // Round up test
+            InsertSingleValue(table, columnName, "2.467");
+            SelectAndValidateValues(table, columnName, SqlDecimal.Parse("2.47"), 1);
+            DeleteFromTable(table, string.Format("{0}={1}", columnName, SqlDecimal.Parse("2.47")), 1);
+            // Round down test
+            InsertSingleValue(table, columnName, "-672.613");
+            SelectAndValidateValues(table, columnName, SqlDecimal.Parse("-672.61"), 1);
+            DeleteFromTable(table, string.Format("{0}={1}", columnName, SqlDecimal.Parse("-672.61")), 1);
+        }
+
+        /// <summary>
+        /// Validates if driver can handle floating point number type correctly
+        /// These Snowflake types are equivalent: FLOAT,FLOAT4,FLOAT8,DOUBLE,DOUBLE PRECISION,REAL
+        /// </summary>
+        [SkippableFact]
+        public void TestFloatNumberRange()
+        {
+            string columnName = "FLOATTYPE";
+            string table = CreateTemporaryTable(string.Format("{0} FLOAT", columnName));
+            ValidateInsertSelectDeleteSingleDoubleValue(table, columnName, 0);
```

Comment:
There are a few ways to structure tests like this. Arrow uses Xunit and InlineData, so we brought that pattern in as well - https://github.com/apache/arrow-adbc/blob/ff222e2004e281d26395964d982320bf5c506160/csharp/test/Apache.Arrow.Adbc.Tests/AdbcTests.cs#L49. Might be worth following this same pattern.

Reply:
> There are a few ways to structure tests like this. Arrow uses Xunit and InlineData, so we brought that pattern in as well -
> 
> https://github.com/apache/arrow-adbc/blob/ff222e2004e281d26395964d982320bf5c506160/csharp/test/Apache.Arrow.Adbc.Tests/AdbcTests.cs#L49
> 
> . Might be worth following this same pattern.

Sorry, I'm not totally clear on this. Do you want me to include status code on return or do you want me to place the value I'm testing first and then the necessary variables afterwards?